### PR TITLE
Format code by clang-format

### PR DESCRIPTION
**Pull request by coder3101**
This commit formats the code by `clang-tidy` according to the styles as mentioned in the `.clang-format` files in the root of the repository. The following locations have been formatted:

- `examples/tensor/*.cpp`
- `test/tensor/*.cpp`
- `include/boost/numeric/ublas/tensor/*.hpp`
- `include/boost/numeric/ublas/tensor/*/*.hpp`

The please review the styles and if any change is required, I will reupdate the files with the new format.

### CODE REVIEW

**Comment by coder3101** [comment]
CI failed because Github actions are down. See https://www.githubstatus.com/

**Comment by coder3101** [comment]
@bassoy and @amitsingh19975 

You can check the code formatting. Finally all CI tests are passing.

File: .github/workflows/code_format.yml [hunk 1]
```diff
@@ -4,21 +4,36 @@
...
+    - name: Commit files
+      run: |
+        git config --local user.email "ublasbot@boost.com"
+        git config --local user.name "ublas-format-bot"
+        git diff --exit-code HEAD || git commit -a --amend --no-edit
+
+    - name: Push changes
+      run: |
+        git remote add token https://${GITHUB_ACTOR}:${{ secrets.GITHUB_TOKEN }}@github.com/${GITHUB_REPOSITORY}.git
+        git push token HEAD:${GITHUB_REF} --force-with-lease=${GITHUB_REF}:${GITHUB_SHA}
```
Comment (coder3101) [inline .github/workflows/code_format.yml:39]: `--force-with-lease` makes sure that forced updates do not overwrite code of other contributors.

**Comment by coder3101** [comment]
@bassoy I have added the auto clang-format feature. Whenever push occurs in our repository, the GitHub action will amend the last commit and format the code force update the repository with the lease.

**Comment by bassoy** [comment]
Thanks, looks good to me.

**Pull request closed without merging**